### ISSUE TITLE
Android Only: ResourceLoader.load_threaded_request() returns ERR_INVALID_PARAMETER Attempted to load a resource already being loaded from this thread

### ISSUE THREAD

**Issue by FeralBytes** [godotengine/godot#76928]
### Godot version

4.0.2.stable

### System information

Android version 12, Samsung SM-A015A OpenGL 3.0.2; Armeabi-v7a

### Issue description

When calling ResourceLoader.load_threaded_request(the_resource_path) on Android it returns error # 31 ERR_INVALID_PARAMETER and generates a debug output from android of:
```
E 0:00:03:0867   load_threaded_request: Attempted to load a resource already being loaded from this thread, cyclic reference?
  <C++ Error>    Method/function failed. Returning: ERR_INVALID_PARAMETER
  <C++ Source>   core/io/resource_loader.cpp:351 @ load_threaded_request()
```
But print output from every step shows that this is not true, this is the first time that I attempted to load the resource.
```
--- DEVICE API >= 21; DEBUGGING OVER USB ---
Reverse result: 0
Godot Engine v4.0.2.stable.official.7a0977ce2 - https://godotengine.org
Vulkan API 1.1.128 - Forward Mobile - Using Vulkan Device #0: Qualcomm - Adreno (TM) 505
 
  BackGroundLoading: A background thread, thread number 1, failed to load resource:'res://EverGreen/Settings.gd' with an error number of 31. ResourceLoader.load_threaded_request() did not return OK.
  BackGroundLoading: A background thread, thread number 1, failed to load resource:'res://EverGreen/Settings.gd' with an error number of 0. ResourceLoader.load_threaded_get_status() did not return THREAD_LOAD_LOADED
```
Of Note: All of the included code works exactly as expected on Linux in the editor and also in a Linux Binary Export.
### Steps to reproduce

Load up the minimal reproduction project 
Install the Android build template and setup the Android export for your device.
And export it to an Android device. 
Then observe that the code fails as stated in the issue description.

### Minimal reproduction project

Thank you all for the Godot Engine and all of your help.
[ResourceLoader.zip](https://github.com/godotengine/godot/files/11446134/ResourceLoader.zip)

**Comment by Calinou** [comment]
- Related to https://github.com/godotengine/godot/issues/75815.

**Comment by FeralBytes** [comment]
@Calinou I know you know way more about these things than I do. But I do not see the connection as Settings.gd does not have any sub resources. It is just a GDScript that is a resource file. Is it still related?

**Comment by Calinou** [comment]
Threaded loading has many issues which should be resolved by https://github.com/godotengine/godot/pull/74405.

**Comment by FeralBytes** [comment]
Awesome, I hope that PR #74405 gets merged, because other wise it seems like background loading on Android is dead in the water. I will wait a few before making everything have to be foreground loading as a work around.

**Comment by akien-mga** [comment]
Possibly fixed by #74405. Please test if you can once 4.1 dev 3 is released.

**Comment by FeralBytes** [comment]
@akien-mga I tested 4.1.dev3 one Android with my previously crashing project, and good news, it now successfully loads using threading, like it is supposed too.
